docs/codebase-best-practices.md

```diff
@@ -0,0 +1,61 @@
+# Codebase Best Practices
+
+## General JavaScript
+
+In most cases, our [linter](how-to-setup-freecodecamp-locally?id=follow-these-steps-to-get-your-development-environment-ready) will warn of any formatting which goes against this codebase's preferred practice.
+
+It is encouraged to use functional components over class-based components.
+
+## Specific TypeScript
+
+### Migrating a JavaScript File to TypeScript
+
+#### Retaining Git File History
+
+Sometimes changing the file from `<filename>.js` -> `<filename>.tsx?` causes the original file to be deleted, and a new one created, and other times the filename just changes - in terms of Git. Ideally, we want the file history to be preserved.
+
+The best bet at achieving this is to:
+
+1. Rename the file
+2. Commit with the flag `--no-verify` to prevent Husky from complaining about the lint errors
+3. Refactor to TypeScript for migration, in a separate commit
+
+> [!NOTE]
+> Editors like VSCode are still likely to show you the file has been deleted and a new one created. If you use the CLI to `git add .`, then VSCode will show the file as renamed in stage
+
+### Naming Conventions
+
+#### Interfaces and Types
+
+For the most part, it is encouraged to use interface over type declarations.
```

Comment:
```suggestion
For the most part, it is encouraged to use `interface` over `type` declarations.
```
Code blocks?

Alternatively:
```suggestion
For the most part, it is encouraged to use interfaces over type declarations.
```
Plural form if not referring to the keyword itself.

Reply:
Yes. I think I was going for code blocks

Reply:
TBH I'd take the verbose option just for clarity.